bin/pygrb/pygrb_make_offline_workflow

```diff
@@ -0,0 +1,370 @@
+parser.add_argument("--version", action="version", version=__version__)
+parser.add_argument("-d", "--output-dir", default=None,
+                    help="Path to output directory.")
+_workflow.add_workflow_command_line_group(parser)
+args = parser.parse_args()
+wflow = _workflow.Workflow(args, workflow_name)
+all_files = _workflow.FileList([])
+tags = []
+
+# Setup run directory
+if args.output_dir:
+    baseDir = args.output_dir
+else:
+    baseDir = os.getcwd()
+runDir = os.path.join(baseDir, "GRB%s" % str(wflow.cp.get("workflow",
+                                                          "trigger-name")))
+if not os.path.exists(runDir):
+    os.makedirs(runDir)
+os.chdir(runDir)
+
+# SEGMENTS
+# Hack to set start and end times based on maximum allowed duration
+start = int(wflow.cp.get("workflow", "trigger-time")) - int(wflow.cp.get(
+        "workflow-exttrig_segments", "max-duration"))
+end = int(wflow.cp.get("workflow", "trigger-time")) + int(wflow.cp.get(
+        "workflow-exttrig_segments", "max-duration"))
+wflow.cp = set_start_end(wflow.cp, start, end)
+
+# Retrieve segments ahope-style
+currDir = os.getcwd()
+segDir = os.path.join(currDir, "segments")
+sciSegs, segsFileList = _workflow.setup_segment_generation(wflow, segDir)
+
+# Make coherent network segments
+onSrc, sciSegs = _workflow.get_triggered_coherent_segment(wflow, segDir,
+                                                          sciSegs)
+# FIXME: The following lines are crude hacks.
+ifo = sciSegs.keys()[0]
+padding = int(wflow.cp.get("inspiral", "pad-data"))
+deadtime = int(wflow.cp.get("inspiral", "segment-duration")) / 4
+wflow.analysis_time = segment(int(sciSegs[ifo][0][0]) + deadtime + padding,
+                              int(sciSegs[ifo][0][1]) - deadtime - padding)
+wflow.cp = set_start_end(wflow.cp, int(sciSegs[ifo][0][0]),
+                         int(sciSegs[ifo][0][1]))
+ext_file = None
+
+# DATAFIND
+dfDir = os.path.join(currDir, "datafind")
+datafind_files, sciSegs = _workflow.setup_datafind_workflow(wflow, sciSegs,
+                                                            dfDir,
+                                                            segsFileList)
+ifo = sciSegs.keys()[0]
+ifos = ''.join(sciSegs.keys())
+
+# If using coh_PTF_inspiral we need bank_veto_bank.xml
+if os.path.basename(wflow.cp.get("executables", "inspiral")) \
+                    == "lalapps_coh_PTF_inspiral":
+    datafind_veto_files = _workflow.FileList([])
+    bank_veto_file = get_coh_PTF_files(wflow.cp, ifos, runDir, bank_veto=True)
+    datafind_veto_files.extend(datafind_files)
+    datafind_veto_files.extend(bank_veto_file)
+    
+    # Make ExtTrig xml file (needed for lalapps_inspinj and summary pages)
+    ext_file = make_exttrig_file(wflow.cp, ifos, sciSegs[ifo][0], baseDir)
+    all_files.extend(_workflow.FileList([ext_file]))
+
+all_files.extend(datafind_veto_files)
+
+# TEMPLATE BANK AND SPLIT BANK
+# TODO: Move from pregenerated to generated coherent network bank
```

Comment:
Wouldn't that also be pregenerated? Or do you want that generation done at run-time? (Would be slow for aligned-spin banks!)